gef.py

```diff
@@ -689,10 +689,33 @@ def size(self) -> int:
             raise AttributeError
         return self.page_end - self.page_start
 
+    def search_for_realpath(self) -> str:
+        """This function is a workaround for gdb bug #23764
+
+        path might be wrong for remote sessions, so try a simple search for files
+        that aren't found at the path indicated, which should be canonical.
+        """
+
+        remote_path = pathlib.Path(self.path)
+        # First, try the canonical path in the remote session root.
+        candidate1 = gef.session.remote.root / remote_path.relative_to(remote_path.anchor)
+        if candidate1.is_file():
+            return str(candidate1)
+
+        # On some systems, /lib(64) might be a symlink to /usr/lib(64), so try removing
+        # the /usr prefix.
+        if remote_path.is_relative_to("/usr"):
+            candidate = gef.session.remote.root / remote_path.relative_to("/usr")
+            if candidate.is_file():
+                return str(candidate)
+
+        # Base case, return the original realpath
+        return str(candidate1)
+
     @property
     def realpath(self) -> str:
```

Comment:
Wouldn't make more sense to make `realpath` a `Path`? Avoiding all the Path <-> str conversions?

Reply:
If we change the type, we have to at least review all of the internal uses, and all of the uses in gef-extras, and we also risk breaking any third-party extensions.  Changing the type would be a breaking change to the API, so I assume that's not something desirable.

But that's your call to make.  Let me know what you'd like to see, here.

Reply:
Fair point, maybe something to work in the future then.